metadata/snapshot.go

```diff
@@ -623,10 +728,25 @@ func (s *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 
 }
 
-func (s *snapshotter) Remove(ctx context.Context, key string) error {
+func (s *snapshotter) Remove(ctx context.Context, key string) (err error) {
 	s.l.RLock()
 	defer s.l.RUnlock()
 
+	defer func() {
```

Comment:
If something fails during `commit` or `prepare` we already have the deferred functions to broadcast, why do we need this here?

Reply:
So the reasoning is that through the entire process only one waiter is ever going through the steps, so we need cleanup on any of the operations that fail. They all currently remove the key if anything goes wrong, so this is the case where something went wrong and we need to Remove the snapshot and try again. If there was no error in Prepare, we won't remove the key, and if we fail *before* Commit then that defer never gets run obviously, so the defers cover the three failure/success cases.

Reply:
(although my logic may be off so please let me know if you find something 🤣)

Reply:
This looks good.